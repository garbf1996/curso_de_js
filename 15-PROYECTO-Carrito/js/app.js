//Varriable 
const lista_curso = document.querySelector('#lista-cursos');
const carrito = document.querySelector('#lista-carrito tbody');
const borrar_curso = document.querySelector('#carrito');
const vaciar_carrito = document.querySelector('#vaciar-carrito');
contenido = [];

//funcion
addClick();
function addClick(){
lista_curso.addEventListener('click',agregar);
borrar_curso.addEventListener('click',borrar);
vaciar_carrito.addEventListener('click',()=>{
  contenido = [];
});
}

function agregar(e){
  e.preventDefault();
  //Haciendo click los contenidos de cursos
if(e.target.classList.contains('agregar-carrito')){
const selec_curso = e.target.parentElement.parentElement;
leerDatos(selec_curso);
}
}

function borrar(e){
if(e.target.classList.contains('borrar-curso')){
const cursoId = e.target.getAttribute('data-id');
contenido = contenido.filter(curso => curso.id !== cursoId);
}
carriHTML();
}

//informacion de los cursos
function leerDatos(curso){

  inforcurso = {
    img:curso.querySelector('img').src,
    titulo:curso.querySelector('h4').textContent,
    precio:curso.querySelector('.precio span').textContent,
    id:curso.querySelector('a').getAttribute('data-id'),
    cantidad : 1,
  }

const exicte = contenido.some(curso=>curso.id === inforcurso.id);

if(exicte){
const  cursos = contenido. map(curso=>{
if(curso.id === inforcurso.id){
curso.cantidad ++;
curso.precio = curso.precio * curso.cantidad;
return curso;
}else{
return curso;
}
contenido = [...cursos]
}); 
}else{
  contenido = [...contenido, inforcurso];
}

  carriHTML();

}

//Carrito
function carriHTML(){
limpiarHtml();
contenido.forEach(curso=>{

  const row = document.createElement('tr');
  row.innerHTML = `
  <td><img src="${curso.img}" width="100"></td>
  <td>${curso.titulo}</td>
  <td>${curso.precio}</td>
  <td>${curso.cantidad}</td>
  <td><a a href="#" class="borrar-curso" data-id="${curso.id}">X</a></td>
  `;
  carrito.appendChild(row);
});
}

function limpiarHtml(){
  while(carrito.firstChild){
    carrito.removeChild(carrito.firstChild);
  }
}


