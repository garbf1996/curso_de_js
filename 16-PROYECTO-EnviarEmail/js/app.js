const  btnEnvair = document.querySelector('#enviar');
const  email = document.querySelector('#email');
const  asunto = document.querySelector('#asunto');
const  mensaje = document.querySelector('#mensaje');
const  formulario = document.querySelector('#enviar-mail');

appStart();
function appStart() {
  document.addEventListener('DOMContentLoaded', btn);
  email.addEventListener('blur', input);
  asunto.addEventListener('blur', input);
  mensaje.addEventListener('blur', input);
}

function btn() {
btnEnvair.disabled = true;
btnEnvair.classList.add('opacity-50','cursor-not-allowed');
}

function input(e){
if(e.target.value.length > 0){
e.target.classList.add('border','border-green-900', 'radius-6')
}else{
  e.target.classList.add('border','border-red-500');
  errorMesje();
}

}


function errorMesje(){

 const ware = document.createElement('p');
 ware.textContent = "Los campos son obligatorios"; 

  ware.classList.add('border','border-red-500', 'background-red-red','text-red-500','p-3','mt-3','text-center', 'error');
const mensaje = document.querySelectorAll('.error');
  
if(mensaje.length === 0){
  formulario.appendChild(ware);
}

}


