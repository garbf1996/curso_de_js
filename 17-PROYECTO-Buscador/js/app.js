const resultado = document.querySelector('#resultado');
const year = document.querySelector('#year');
const marca = document.querySelector('#marca');


const datosBusqueda = {
    marca: '',
    modelo: '',
    year: '',
    minimo: '',
    maximo: '',
    puertas: '',
    transmision: '',
    color: ''
}

const max = new Date().getFullYear();
const min = max -20;

eventos();
function eventos(){
    document.addEventListener('DOMContentLoaded',()=>{
        mostarAuto(autos);
        select();
    })

    marca.addEventListener('change',e=>{
        datosBusqueda.marca = e.target.value;
    });
}

function mostarAuto(autos){

autos.forEach(autos => {
const autoHtml = document.createElement('p');
const{marca,modelo,year,puertas,transmision,precio,color} = autos;
autoHtml.textContent =`
${marca}  -${modelo} -${year} -Puertas -${puertas} -Transmision ${transmision} -Precio ${precio} -Color ${color}
`;
resultado.appendChild(autoHtml);
});
}
//Funcion para el ciclo de year
function select(){
 for(let i = max; i>= min; i--){
 const option = document.createElement('option');
 option.value = i;
 option.textContent = i
 year.appendChild(option);
 }
}

function filtraAuto(){
    const resultado = autos.filter(filtramarca);

    if(resultado.length){
        mostarAuto(resultado);
    }else{
        console.log("no hay resultado");
    }
}

function filtramarca(){
    const {marca} =  datosBusqueda;
    if(marca){
     return autos.marca === marca;
    }
    return autos;
}