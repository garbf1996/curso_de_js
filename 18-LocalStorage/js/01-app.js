 
const persona = [
 nombre='Garber',
 pellidos='Batista',
 edad = 25   
];

localStorage.setItem('persona', JSON.stringify(persona));