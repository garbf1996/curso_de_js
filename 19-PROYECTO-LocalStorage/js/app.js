const lista_tweets  = document.querySelector('#lista-tweets');
const formulario = document.querySelector('#formulario');

let tweets = [];

eventos();
function eventos(){
     formulario.addEventListener('submit', enviamdoTweets);
      document.addEventListener('DOMContentLoaded', ()=>{
       tweets = JSON.parse(localStorage.getItem('tweets')); 
       console.log(tweets);
       tweetsHTML();
     })
}


function enviamdoTweets(e){
e.preventDefault();
const tweet = document.querySelector('#tweet').value;
if(tweet === ''){
MostrarError('No tweet found');
 return;
}

const tweetObj = {
    id:Date.now(),
    tweet
}

tweets = [...tweets, tweet];

tweetsHTML();
formulario.reset();

}

function MostrarError(error){
 const mesajeError = document.createElement('p');
 mesajeError.textContent =error;
 mesajeError.classList.add('error');

const contenido = document.querySelector('#contenido');
contenido.appendChild(mesajeError);

setTimeout(() => {
    mesajeError.remove();
}, 3000);

}

function tweetsHTML(){
clearHTML();
    if(tweets.length > 0){
        tweets.forEach(tweet => {
         const li = document.createElement('li');
         li.textContent = tweet;
         lista_tweets.appendChild(li);
        });
    }
        sincronizarStorage();
}

function sincronizarStorage() {
    localStorage.setItem('tweets', JSON.stringify(tweets));
}

function clearHTML(){
while (lista_tweets.firstChild) {
    lista_tweets.removeChild(lista_tweets.firstChild);
}
}

