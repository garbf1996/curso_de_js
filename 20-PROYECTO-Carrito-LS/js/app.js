const lista_cursos = document.querySelector('#lista-cursos');
const lista_carrito = document.querySelector('#lista-carrito tbody');
const carrito = document.querySelector('#carrito');
const vaciar_carrito = document.querySelector('#vaciar-carrito');
contenidos = [];


eventos();
function eventos(){
lista_cursos.addEventListener('click', comprarCurso);
carrito.addEventListener('click', quitar_lista);
vaciar_carrito.addEventListener('click', ()=>{
    contenidos =[];
});

document.addEventListener('DOMContentLoaded', ()=>{
    contenidos = JSON.parse(localStorage.getItem('carrito'));
    carritoHTML();
});
}


function quitar_lista(e){
e.preventDefault();

if(e.target.classList.contains('borrar-curso')){
 const cursoID = e.target.getAttribute('data-id');
contenidos = contenidos.filter(cursor => cursor.id !== cursoID);
}
carritoHTML();
}

function comprarCurso(e){
e.preventDefault();
if(e.target.classList.contains('agregar-carrito')){
const selec_curso = e.target.parentElement.parentElement;
leeDatos(selec_curso);
}
}

function leeDatos(curso){

    inforCurso = {
        img:curso.querySelector('img').src,
        titulo:curso.querySelector('h4').textContent,
        precio:curso.querySelector('.precio span').textContent,
        id:curso.querySelector('a').getAttribute('data-id'),
        cantidad : 1
    }

    const exicte = contenidos.some(curso => curso.id === inforCurso.id);

    if(exicte){
       
    const cursos = contenidos.map(curso =>{

    if(curso.id === inforCurso.id){
    curso.cantidad ++;
     return curso;
    }else{
     return curso;
    }

    carrito = [...cursos];

    });

    }else{
        contenidos = [...contenidos,inforCurso];
    }
    carritoHTML();
}


function carritoHTML(){
limpaiHTML();
contenidos.forEach(curso => {

const row = document.createElement("tr");    
row.innerHTML = `
<td><img src="${curso.img}" width="100"></td>
<td>${curso.titulo}</td>
<td>${curso.precio}</td>
<td>${curso.cantidad}</td>
<td> <a href="#"  class="borrar-curso" data-id="${curso.id}"</a>x</td>
`;

lista_carrito.appendChild(row);

});
local();
}

function local (){
    localStorage.setItem("carrito", JSON.stringify(contenidos));
}

function limpaiHTML() {
  while (lista_carrito.firstChild) {
    lista_carrito.removeChild(lista_carrito.firstChild);
  }
}