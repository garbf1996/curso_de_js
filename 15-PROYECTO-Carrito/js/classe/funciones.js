import {lista_carrito} from'./selectores.js';

document.addEventListener('DOMContentLoaded',()=>{
    if(localStorage.getItem('carrito')){
        articulo = JSON.parse(localStorage.getItem('carrito'));
        carritoHTML();
    }
});
let  articulo = [];
// Seleciona los cursos
export function agregarCarrito(e){
    e.preventDefault();
   if(e.target.classList.contains('agregar-carrito')){
       const cursoSelect = e.target.parentElement.parentElement;
      leerDatos(cursoSelect);
   }

}
// Lee los datos del curso
function leerDatos(curso){
    const infoCurso = {
        imagen: curso.querySelector('img').src,
        titulo: curso.querySelector('h4').textContent,
        precio: curso.querySelector('.precio span').textContent,
        id: curso.querySelector('a').getAttribute('data-id'),
        cantidad: 1
    }
     
    const existe = articulo.some(curso => curso.id === infoCurso.id);
    if(existe){
    const cursos = articulo.map(curso => {
        if(curso.id === infoCurso.id){
            curso.cantidad++;
            return curso;
        }else{
            return curso;
        }
        articulo = [...cursos];
    });    
    }else{
    articulo = [...articulo,infoCurso];
    }
    carritoHTML();
}
// Muestra el carrito en el HTML
export function carritoHTML(){
limpairHTML();
    articulo.forEach(curso => {
        const row = document.createElement('tr');
        row.innerHTML = `
        <td>
            <img src="${curso.imagen}" width=100>
        </td>
        <td>${curso.titulo}</td>
        <td>${curso.precio}</td>
        <td>${curso.cantidad}</td>
        <td>
            <a href="#" class="borrar-curso" data-id="${curso.id}">X</a>
        </td>
        `;
        lista_carrito.appendChild(row);
    });

    localStorageCarrito();

}
// Limpia el HTML
function limpairHTML(){
    while(lista_carrito.firstChild){
        lista_carrito.removeChild(lista_carrito.firstChild);
    }
}
// Elimina un curso del carrito
export function borrarCurso(e){
    e.preventDefault();
    if(e.target.classList.contains('borrar-curso')){
        const cursoId = e.target.getAttribute('data-id');
        articulo = articulo.filter(curso => curso.id !== cursoId);
        carritoHTML();
    }
}


// Vacia el carrito
export function cleaCurso(e){
    e.preventDefault();
    articulo = [];
    carritoHTML();
}

 function localStorageCarrito(){
    localStorage.setItem('carrito',JSON.stringify(articulo));
}