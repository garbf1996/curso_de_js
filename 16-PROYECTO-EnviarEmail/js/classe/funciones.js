import {enviarBTN,rc} from './selectores.js'

export function appStar(){
    enviarBTN.disabled = true;
    enviarBTN.classList.add('cursor-not-allowed');
    }
    

export function validar(e){
    //Confimar si los compos es llenos
    if(e.target.value.length > 0){
      e.target.classList.add('border','border-green-500');
    }else{
        e.target.classList.add('border','border-red-500');
    }
    //validar si es un correo
    if(e.target.type == 'email'){
       if(rc.test(e.target.value)){
        e.target.classList.add('border','border-green-500');
       }else{
        e.target.classList.add('border','border-red-500');
       }
    }
}


