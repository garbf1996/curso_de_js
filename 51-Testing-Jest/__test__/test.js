import {suma} from '../js/funciones.js';

describe('Testing a las funciones de suma y reta',()=>{
    test ('Suma de 20 y 30', ()=>{
       expect (suma(20,30)).toBe(50)
    });
 });
