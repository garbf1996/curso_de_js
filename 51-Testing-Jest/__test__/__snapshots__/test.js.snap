// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing al cliente Es Garber Batista 1`] = `
{
  "balance": 500,
  "nombre": "Ashaly Batista",
  "tipo": "Premium",
}
`;
